dask/array/tests/test_reductions.py

```diff
@@ -21,6 +21,26 @@ def assert_eq(a, b):
     _assert_eq(a, b, equal_nan=True)
 
 
+def assert_nan_eq(a, b):
+    """ Similar to assert_eq but accept nan value """
+    if hasattr(a, 'compute'):
+        a = a.compute()
+    if hasattr(b, 'compute'):
+        b = b.compute()
+
+    assert type(a) == type(b)
+    a = np.array(a)
+    b = np.array(b)
+    assert a.shape == b.shape
+
+    nanidx = a != b
+    assert np.isnan(a[nanidx].astype(float)).all()
+    assert np.isnan(b[nanidx].astype(float)).all()
+    a[nanidx] = 0.0
+    b[nanidx] = 0.0
+    assert_eq(a, b)
```

Comment:
If possible I'm somewhat in favor of including `nan` handling logic directly into `assert_eq`.  We've tried to centralize all testing logic into that one function in the past for the sake of simplicity.  I can't think of a case where I would prefer to have non-nan-handling behavior.  We could make this a keyword argument if desired.

Reply:
This isn't needed, adding `equal_nan=True` to `assert_eq` already works.